torchmdnet/models/model.py

```diff
@@ -426,3 +430,37 @@ def forward(
         # Returning an empty tensor allows to decorate this method as always returning two tensors.
         # This is required to overcome a TorchScript limitation, xref https://github.com/openmm/openmm-torch/issues/135
         return y, torch.empty(0)
+
+
+class Ensemble(torch.nn.ModuleList):
+    """Average predictions over an ensemble of TorchMD-Net models"""
+
+    def __init__(self, modules):
+        super().__init__(modules)
+
+    def forward(
+        self,
+        z: Tensor,
+        pos: Tensor,
+        batch: Optional[Tensor] = None,
+        box: Optional[Tensor] = None,
+        q: Optional[Tensor] = None,
+        s: Optional[Tensor] = None,
+        extra_args: Optional[Dict[str, Tensor]] = None,
+    ):
+        y = []
+        neg_dy = []
+        for model in self:
+            res = model(
+                z=z, pos=pos, batch=batch, box=box, q=q, s=s, extra_args=extra_args
+            )
+            y.append(res[0])
+            neg_dy.append(res[1])
+
+        y = torch.stack(y)
+        neg_dy = torch.stack(neg_dy)
+        y_mean = torch.mean(y, axis=0)
+        neg_dy_mean = torch.mean(neg_dy, axis=0)
+        y_std = torch.std(y, axis=0)
+        neg_dy_std = torch.std(neg_dy, axis=0)
+        return y_mean, neg_dy_mean, y_std, neg_dy_std
```

Comment:
You could change it to `{"y": {"mean": y_mean, "std": y_std}, "neg_dy":{"mean": neg_dy_mean, "std": neg_dy_std}}` 
It would be interesting to have this class be somehow interface-compatible with TorchMD_Net, but you are returning 4 tensors anyway, so it cannot be.

Reply:
I think I prefer it as is in the sense that it keeps compatibility in the first 2 returned arguments. So all models return as first the y and second the neg_dy and this one does too. And the other args are extra information. Converting the result to a dict will change more the style I think.

Reply:
I think if I can add an argument to load_model called `return_std=True` then it would be clearer when the Ensemble returns two or 4 args. I'll do that

Reply:
I implemented this now. Give it another look